### ISSUE TITLE
Bug: Maximum call stack size exceeded when checking immutability

### ISSUE THREAD

**Issue by flysand7** [eslint/eslint#17742]
### Environment

Node version: dont have node
npm version: dont have npm
Local ESLint version: v8.53.0
Global ESLint version: not sure what this is for
Operating System: `Linux a2c7bfed2354 6.2.0-35-generic #35~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Fri Oct  6 10:23:26 UTC 2 x86_64 GNU/Linux`

### What parser are you using?

@typescript-eslint/parser

### What did you do?

```js

I'm pretty sure this is the relevant code. Using zod `3.22.4`

```
import { z } from "zod";

export const GetPaginationT = z.object({
  limit: z.number(),
  page:  z.number(),
});

export type GetPagination = z.infer<typeof GetPaginationT>
```
```

### What did you expect to happen?

Something that doesn't cause the linter to fail on apparently-valid types...

### What actually happened?

Crash. The error log:

`typeArgumentStrings` contains `undefined`, this is likely a bug in `is-immutable-type`

Oops! Something went wrong! :(

ESLint: 8.53.0

RangeError: Maximum call stack size exceeded.
Occurred while linting /app/src/model/dto/index.ts:24
Rule: "functional/type-declaration-immutability"
    at get (/app/node_modules/ts-api-utils/lib/index.cjs:15:121)
    at getTypeData (/app/node_modules/is-immutable-type/dist/index.cjs:301:18)
    at <anonymous> (/app/node_modules/is-immutable-type/dist/index.cjs:248:43)
    at map (:1:11)

### Link to Minimal Reproducible Example

don't have time for that. maybe later, maybe never

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

I'm not experienced in typescript, not really my area of expertise.

**Comment by Rec0iL99** [comment]
Hi @flysand7, as the bug report template states, we do require a reproducible example for bug reports to better debug the issue and save time. 

I'm closing this since the error is coming from a plugin that the ESLint team does not maintain. Please open an issue there.

**Comment by flysand7** [comment]
Ah sorry, wrong repository I guess
